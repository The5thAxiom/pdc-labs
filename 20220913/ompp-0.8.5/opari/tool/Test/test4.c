#include <stdio.h>

int main() {
  int i;

  #pragma omp parallel
  {
    printf("parallel\n");
    #pragma omp for nowait
    for (i=0; i<4; ++i) printf("do ; ; ; nowait %d\n", i);

    #pragma omp barrier

    #pragma omp \
	   /* unnütz */ \
	   /* mehr unnütz */ \
	   for /* KEIN nowait */
    for (i=0; i<4; ++i) {
      printf("do nowait? %d\n", i);
    }

    #pragma omp for
    for (i=0; i<4; ++i) {
      printf("do %d\n", i);
    }
  }
}
