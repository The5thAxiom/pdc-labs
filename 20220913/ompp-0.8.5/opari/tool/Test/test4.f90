       program test4
       integer i

       !$omp parallel
       write(*,*) "parallel"
         !$omp do
         do i=1,4
           write(*,*) "do nowait",i
         enddo
         !$omp enddo nowait

         !$omp barrier

         !$omp do
         do i=1,4
           write(*,*) "do nowait?",i
         enddo
         !$omp   end   &  ! continue
         !$omp &       &  ! still continue
         !$omp & do       ! kein nowait

         !$omp do
         do i=1,4
           write(*,*) "do",i
         enddo
         !$omp end do
       !$omp end parallel

       end
