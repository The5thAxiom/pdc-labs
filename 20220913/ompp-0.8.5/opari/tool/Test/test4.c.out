#ifdef _POMP
#  undef _POMP
#endif
#define _POMP 200110

#include "test4.c.opari.inc"
#line 1 "test4.c"
#include <stdio.h>

int main() {
  int i;

POMP_Parallel_fork(&omp_rd_16);
#line 6 "test4.c"
  #pragma omp parallel POMP_DLIST_00016
{ POMP_Parallel_begin(&omp_rd_16);
#line 7 "test4.c"
  {
    printf("parallel\n");
POMP_For_enter(&omp_rd_17);
#line 9 "test4.c"
    #pragma omp for nowait
    for (i=0; i<4; ++i) printf("do ; ; ; nowait %d\n", i);
POMP_For_exit(&omp_rd_17);
#line 11 "test4.c"

POMP_Barrier_enter(&omp_rd_18);
#line 12 "test4.c"
    #pragma omp barrier
POMP_Barrier_exit(&omp_rd_18);
#line 13 "test4.c"

POMP_For_enter(&omp_rd_19);
#line 14 "test4.c"
    #pragma omp \
	                \
	                     \
	   for                   nowait
    for (i=0; i<4; ++i) {
      printf("do nowait? %d\n", i);
    }
POMP_Barrier_enter(&omp_rd_19);
#pragma omp barrier
POMP_Barrier_exit(&omp_rd_19);
POMP_For_exit(&omp_rd_19);
#line 21 "test4.c"

POMP_For_enter(&omp_rd_20);
#line 22 "test4.c"
    #pragma omp for nowait
    for (i=0; i<4; ++i) {
      printf("do %d\n", i);
    }
POMP_Barrier_enter(&omp_rd_20);
#pragma omp barrier
POMP_Barrier_exit(&omp_rd_20);
POMP_For_exit(&omp_rd_20);
#line 26 "test4.c"
  }
POMP_Barrier_enter(&omp_rd_16);
#pragma omp barrier
POMP_Barrier_exit(&omp_rd_16);
POMP_Parallel_end(&omp_rd_16); }
POMP_Parallel_join(&omp_rd_16);
#line 27 "test4.c"
}
